WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/notifications/NotificationsFragment.kt

```diff
@@ -1,19 +1,250 @@
+        return FragmentNotificationsBinding.inflate(layoutInflater)
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         setHasOptionsMenu(true)
         super.onCreate(savedInstanceState)
     }
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        observeUiInteractions()
 
+    }
+    /**
+     * Vibrates the device. Used for providing feedback when the user performs an action.
+     */
+    fun vibrate(view: View) {
+        view.performHapticFeedback(HapticFeedbackConstants.LONG_PRESS)
+    }
+    private fun defaultLed(){
+        val notificationManager= context?.getSystemService(NOTIFICATION_SERVICE) as NotificationManager
 
+        val notification_Channel_ID = "my_channel_id_01"
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationChannel = NotificationChannel(
+                notification_Channel_ID,
+                "My Notifications",
+                NotificationManager.IMPORTANCE_HIGH
+            )
+
+            // Configure the notification channel.
+
+            notificationChannel.description = "Channel description"
+            notificationChannel.enableLights(true)
+            notificationChannel.lightColor = Color.RED
+            notificationChannel.vibrationPattern = longArrayOf(0, 1000, 500, 1000)
+            notificationChannel.enableVibration(true)
+            notificationManager.createNotificationChannel(notificationChannel)
+        }
+
+    }
+    private fun observeUiInteractions() {
+        //Message Layouts
+        binding.vibrateLayout.setOnClickListener {
+            val builder = AlertDialog.Builder(context)
+            builder.setTitle("Vibrate")
+            val vibrates = arrayOf("Off", "Default", "Short", "Long")
+            val checkedItem = 1
+            builder.setSingleChoiceItems(vibrates, checkedItem) { dialog, which ->
+                val i = (dialog as AlertDialog).listView.checkedItemPosition
+                if (i == 0) {
+                    binding.vibrateDesc.text = getString(R.string.off)
+                } else if (i == 1) {
+                    binding.vibrateDesc.text = getString(R.string.defaultanswer)
+                } else if (i == 2) {
+                    binding.vibrateDesc.text = getString(R.string.tvshort)
+                } else if (i == 3) {
+                    binding.vibrateDesc.text = getString(R.string.tvlong)
+                }
+                dialog.dismiss()
+            }
+            val dialog = builder.create()
+            dialog.show()
+        }
+        binding.popUpLayout.setOnClickListener {
+            val builder = AlertDialog.Builder(context)
+            builder.setTitle("Popup notification")
+            val popup = arrayOf("No popup", "Only when screen is 'on' ", "Only when screen is 'off'", "Always show popup")
+            val checkedItem = 1
```

Comment:
should be 0 right? if thats the default

Reply:
number doesnt matter too much it will be changed when the implementation of the feature gets added